scripts/generate-assets.js

```diff
@@ -29,32 +29,46 @@ if (!fs.existsSync(baseDir)) {
 const target = path.join(baseDir, `assets.json`);
 const assets = {};
 
+console.log('1');
 (async () => {
-  await processDirectory(assets, path.join(__dirname, '../rulesets/oas'));
-  await writeFileAsync(target, JSON.stringify(assets, null, 2));
+  for (const kind of ['oas', 'aas']) {
+    console.log('2');
+    await processDirectory(assets, path.join(__dirname, `../rulesets/${kind}`));
+    console.log('3');
+    await writeFileAsync(target, JSON.stringify(assets, null, 2));
+    console.log('4');
+  }
 })();
 
 async function processDirectory(assets, dir) {
   await Promise.all((await readdirAsync(dir)).map(async name => {
+    console.log(name);
     if (name === 'schemas') return;
     const target = path.join(dir, name);
     const stats = await statAsync(target);
     if (stats.isDirectory()) {
       return processDirectory(assets, target);
     } else {
       let content = await readFileAsync(target, 'utf8');
+      console.log('pro ' + target)
       if (path.extname(name) === '.json') {
-        content = JSON.stringify((await httpAndFileResolver.resolve(JSON.parse(content), {
+        console.log('awaiting ' + target)
+        const x = await httpAndFileResolver.resolve(JSON.parse(content), {
           dereferenceRemote: true,
           dereferenceInline: false,
           baseUri: target,
           parseResolveResult(opts) {
             opts.result = parse(opts.result);
+            console.log('parsed ' + opts.targetAuthority.toString());
+
             return opts;
           },
-        })).result);
+        })
+        console.log('awaited ' + target)
+        content = JSON.stringify((x).result);
+        console.log('done ' + target);
```

Comment:
@P0lip I might need help here as I'm hitting a wall. `generate-assets` eventually makes the build timeout.

My lame attempt at trying to understand the issue led me to this.

![image](https://user-images.githubusercontent.com/92363/75291206-1143a680-5822-11ea-9ab3-f9cfdc095601.png)

The resolver seems to hit some infinite loop while processing the json schema draft 7 (that asyncapi depends on)

![image](https://user-images.githubusercontent.com/92363/75291535-b2326180-5822-11ea-86ae-4bde59ca1f95.png)

Any idea? (please, say "Yes").

Reply:
Urgh, I think I know what the issue is. This is something Phil faced some time ago, see
https://github.com/stoplightio/spectral/issues/789. I believe the most reasonable solution would be to use json-schema-ref-parser for dereferencing

Reply:
@P0lip On it. Thanks! <3

Reply:
@P0lip Thanks to you, I went a step farther. However, I need to get a better understanding of the overall constraints governing the use of those static assets:

 - Are they only used in Spectral test environments?
 - Are there some parts of the Studio ecosystem that leverage the `Spectral.registerStaticAssets()` method?

Reply:
@P0lip @philsturgeon 🎉 At last! I got it. Took some time.

`json-schema-ref-parser` ended up being a dead-end because of an issue I've hit. I'll open an issue upstream.

I had to slightly "hack" the vendored aas2 schema. This allowed me to reuse most of the current code with minimal changes (I had to inline this **[part](https://github.com/asyncapi/asyncapi/blob/master/versions/2.0.0/schema.json#L358)**).